src/metatrain/utils/scaler.py

```diff
@@ -0,0 +1,237 @@
+
+            self.scales[self.output_name_to_output_index[target_key]] = np.sqrt(
+                sum_of_squared_targets / total_num_elements
+            )
+
+    def restart(self, dataset_info: DatasetInfo) -> "Scaler":
+        # merge old and new dataset info
+        merged_info = self.dataset_info.union(dataset_info)
+
+        self.new_targets = {
+            key: value
+            for key, value in merged_info.targets.items()
+            if key not in self.dataset_info.targets
+        }
+
+        # register new outputs
+        for target_name, target in self.new_targets.items():
+            self._add_output(target_name, target)
+
+        self.dataset_info = merged_info
+
+        return self
+
+    def forward(
+        self,
+        outputs: Dict[str, TensorMap],
+    ) -> Dict[str, TensorMap]:
+        """
+        Scales all the targets in the outputs dictionary back to their
+        original scale.
+
+        :param outputs: A dictionary of target quantities and their values
+            to be scaled.
+
+        :raises ValueError: If an output does not have a corresponding
+            scale in the scaler model.
+        """
+
+        for output_name in outputs:
+            if output_name.startswith("mtt::aux::") or output_name == "features":
+                continue
+            if output_name not in self.outputs.keys():
+                raise ValueError(
+                    f"output key {output_name} is not supported by this scaler "
+                    "model."
+                )
+
+        scaled_outputs: Dict[str, TensorMap] = {}
+        for target_key, target in outputs.items():
+            if target_key.startswith("mtt::aux::") or target_key == "features":
+                scaled_outputs[target_key] = target
+            else:
+                scale = float(
+                    self.scales[self.output_name_to_output_index[target_key]].item()
+                )
+                scaled_target = metatensor.torch.multiply(target, scale)
+                scaled_outputs[target_key] = scaled_target
+
+        return scaled_outputs
+
+    def _add_output(self, target_name: str, target_info: TargetInfo) -> None:
+
+        self.outputs[target_name] = ModelOutput(
+            quantity=target_info.quantity,
+            unit=target_info.unit,
+            per_atom=True,
+        )
+
+        self.scales = torch.cat(
+            [self.scales, torch.tensor([1.0], dtype=self.scales.dtype)]
+        )
+        self.output_name_to_output_index[target_name] = len(self.scales) - 1
+
+    def get_scales_dict(self) -> Dict[str, float]:
+        """
+        Return a dictionary with the scales for each output and output gradient.
+
+        :return: A dictionary with the scales for each output and output gradient.
+            These correspond to the standard deviation of the targets in the
+            original dataset. The scales for each output gradient are the same
+            as the corresponding output.
```

Comment:
Do we need to have separate dict entries for the gradients if the values are always the same (as they should be)?

Reply:
It's nice to have them when you call from outside. Yes, they are the same. If callers of the function want to ignore the gradients, they can do it and use only the values from the base quantities